policies/templates/gcp_iam_restrict_service_account_key_age_v1.yaml

```diff
@@ -0,0 +1,76 @@
+# Copyright 2019 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: templates.gatekeeper.sh/v1alpha1
+kind: ConstraintTemplate
+metadata:
+  name: gcp-iam-restrict-service-account-key-age-v1
+spec:
+  crd:
+    spec:
+      names:
+        kind: GCPIAMRestrictServiceAccountKeyAgeConstraintV1
+        plural: gcpiamrestrictserviceaccountkeyageconstraintsv1
+      validation:
+        openAPIV3Schema:
+          properties:
+            max_age:
+              type: string
+              description: "Duration (e.g. 2400h)"
+  targets:
+   validation.gcp.forsetisecurity.org:
+      rego: | #INLINE("validator/gcp_iam_restrict_service_account_key_age.rego")
+          # Copyright 2019 Google LLC
+          #
+          # Licensed under the Apache License, Version 2.0 (the "License");
+          # you may not use this file except in compliance with the License.
+          # You may obtain a copy of the License at
+          #
+          #      http://www.apache.org/licenses/LICENSE-2.0
+          #
+          # Unless required by applicable law or agreed to in writing, software
+          # distributed under the License is distributed on an "AS IS" BASIS,
+          # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+          # See the License for the specific language governing permissions and
+          # limitations under the License.
+          #
+          
+          package templates.gcp.GCPIAMRestrictServiceAccountKeyAgeConstraintV1
+          
+          import data.validator.gcp.lib as lib
+          
+          deny[{
+          	"msg": message,
+          	"details": metadata,
+          }] {
+          	constraint := input.constraint
+          	asset := input.asset
+          	asset.asset_type == "iam.googleapis.com/ServiceAccount"
```

Comment:
service account doesn't contain the key information, where did you get the reference/test data from?
https://cloud.google.com/iam/reference/rest/v1/projects.serviceAccounts#ServiceAccount

Do you mean serviceaccountkey?